src/app/+collection-page/edit-collection-page/collection-source/collection-source.component.html

```diff
@@ -0,0 +1,56 @@
+<div class="container-fluid">
+  <div class="d-inline-block float-right">
+    <button class=" btn btn-danger" *ngIf="!(isReinstatable() | async)"
+            [disabled]="!(hasChanges() | async)"
+            (click)="discard()"><i
+      class="fas fa-times"></i>
+      <span class="d-none d-sm-inline">&nbsp;{{"item.edit.metadata.discard-button" | translate}}</span>
+    </button>
+    <button class="btn btn-warning" *ngIf="isReinstatable() | async"
+            (click)="reinstate()"><i
+      class="fas fa-undo-alt"></i>
+      <span class="d-none d-sm-inline">&nbsp;{{"item.edit.metadata.reinstate-button" | translate}}</span>
+    </button>
+    <button class="btn btn-primary" [disabled]="!(hasChanges() | async) || !isValid()"
+            (click)="onSubmit()"><i
+      class="fas fa-save"></i>
+      <span class="d-none d-sm-inline">&nbsp;{{"item.edit.metadata.save-button" | translate}}</span>
+    </button>
+  </div>
+  <h4>{{ 'collection.edit.tabs.source.head' | translate }}</h4>
+  <div *ngIf="contentSource" class="form-check mb-4">
+    <input type="checkbox" class="form-check-input" id="externalSourceCheck" [checked]="(contentSource?.harvestType !== harvestTypeNone)" (change)="changeExternalSource()">
```

Comment:
If a collection doesn't have any defined OAI harvesting source, if you click this checkbox, don't do any changes and then click again, the "Save" option will be available and you can save that record. When I did it, I got this error: 
Server Error
Cannot read property '_links' of null

![2019-11-06 10_06_34-Collection Edit - Content Source](https://user-images.githubusercontent.com/877488/68290338-b114d300-007f-11ea-872a-0b82efcc0d3d.png)


client.js:211618 TypeError: Cannot read property '_links' of null
    at DSpaceRESTv2Serializer../src/app/core/dspace-rest-v2/dspace-rest-v2.serializer.ts.DSpaceRESTv2Serializer.deserialize (client.js:218542)
    at ContentSourceResponseParsingService../src/app/core/data/content-source-response-parsing.service.ts.ContentSourceResponseParsingService.parse (client.js:214335)
    at MapSubscriber.project (client.js:217214)
    at MapSubscriber../node_modules/rxjs/_esm5/internal/operators/map.js.MapSubscriber._next (client.js:177922)
    at MapSubscriber../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next (client.js:172017)
    at

Reply:
Since it doesn't make sense for the "Save" button to be enabled in this case, I've added an extra check to disable the button.